src/DSharp.Compiler/CodeModel/ParseNodeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace DSharp.Compiler.CodeModel
+{
+    internal static class ParseNodeExtensions
+    {
+        public static T As<T>(this ParseNode self)
+            where T : ParseNode
+        {
+            return (T)self;
+        }
+
+        public static T FindParent<T>(this ParseNode self)
+            where T : ParseNode
+        {
+            var current = self.Parent;
+
+            while (current != null)
+            {
+                if(current is T)
+                {
+                    return (T)current;
```

Comment:
I know this is bind picky but can we use out new method is As<T>? just to make it consistent and the only possible failure point